src/main/kotlin/uk/gov/dluhc/emsintegrationapi/service/PostalVoteApplicationService.kt

```diff
@@ -73,6 +71,38 @@ class PostalVoteApplicationService(
         )
     }
 
+    private fun doReceipt(request: EMSApplicationResponse, postalVoteApplication: PostalVoteApplication) {
+        with(postalVoteApplication) {
+            status = RecordStatus.DELETED
+            updatedBy = SourceSystem.EMS
+            with(applicationDetails) {
+                emsStatus = when (request.status) {
+                    EMSApplicationStatus.SUCCESS -> ApplicationDetails.EmsStatus.SUCCESS
+                    EMSApplicationStatus.FAILURE -> ApplicationDetails.EmsStatus.FAILURE
+                }
+                emsMessage = request.message
+                emsDetails = request.details
+            }
+        }
+
+        postalVoteApplicationRepository.saveAndFlush(postalVoteApplication)
```

Comment:
interested to know why you need a flush here

Reply:
Removed, no need for flush.